custom_typings/index.d.ts

```diff
@@ -0,0 +1,22 @@
+
+declare module 'custom_typings' {
+    import * as vscode from 'vscode';
+
+    export type TImage = {
+        id: string, // hash256(imageUri.path), e.g. hash256("/c:/Users/image.jpg")
+        uri: vscode.Uri,
+        ext: string, // file extension in upper case, e.g. "JPG"
+        size: number,
+        mtime: number,
+        ctime: number,
+        status: "" | "refresh",
+    };
+
+    export type TFolder = {
+        id: string, // hash256(folderUri.path), e.g. hash256("/c:/Users")
+        path: string, // folderUri.path, e.g. "/c:/Users"
+        images: {
+            [imageId: string]: TImage,
+        },
+    };
+}
```

Comment:
Why are we putting `custom_typings` directly under the root directory? Is this the standard practice? I feel like having all source code placed under `src` seems more reasonable.

Reply:
Not sure, let me try putting it in `src`